FBRetainCycleDetector/FBRetainCycleUtils.m

```diff
@@ -19,9 +19,27 @@
 #import "FBObjectiveCObject.h"
 #import "FBObjectGraphConfiguration.h"
 
-FBObjectiveCGraphElement *FBWrapObjectGraphElementWithContext(id object,
+static BOOL _ShouldBreakGraphEdge(FBObjectGraphConfiguration *configuration,
+                                  FBObjectiveCGraphElement *fromObject,
+                                  NSString *byIvar,
+                                  Class toObjectOfClass) {
+  for (FBGraphEdgeFilterBlock filterBlock in configuration.filterBlocks) {
+    if (filterBlock(fromObject, byIvar, toObjectOfClass) == FBGraphEdgeInvalid) {
+      return YES;
+    }
+  }
+
+  return NO;
+}
+
+FBObjectiveCGraphElement *FBWrapObjectGraphElementWithContext(FBObjectiveCGraphElement *sourceElement,
+                                                              id object,
                                                               FBObjectGraphConfiguration *configuration,
                                                               NSArray<NSString *> *namePath) {
+  if (_ShouldBreakGraphEdge(configuration, sourceElement, [namePath objectAtIndex:0], object_getClass(object))) {
+    return nil;
```

Comment:
Just a thought: Instead of returning `nil`, could this return an "empty" `FBObjectiveCGraphElement`? Something that has no edges, if that even makes sense? Then the places that call this method don't have to check for `nil` before adding to collections.